[BUG] join_test.py::test_sortmerge_join_with_conditionals failed in databricks 8.2 runtime

**Describe the bug**
test added in https://github.com/NVIDIA/spark-rapids/pull/3089, looks like not include GpuSortMergeJoinExec.scala change for spark311db

Initially saw tests failed in databricks 8.2 runtime, `Caused by: java.lang.IllegalStateException: Conditional joins are not supported on the GPU`. Failed tests list,
```bash
[2021-07-30T04:53:04.421Z] FAILED ../../src/main/python/join_test.py::test_sortmerge_join_with_conditionals[Timestamp][IGNORE_ORDER({'local': True})]
[2021-07-30T04:53:04.421Z] FAILED ../../src/main/python/join_test.py::test_sortmerge_join_with_conditionals[Null][IGNORE_ORDER({'local': True})]
[2021-07-30T04:53:04.421Z] FAILED ../../src/main/python/join_test.py::test_sortmerge_join_with_conditionals[Float][IGNORE_ORDER({'local': True}), INCOMPAT]
[2021-07-30T04:53:04.421Z] FAILED ../../src/main/python/join_test.py::test_sortmerge_join_with_conditionals[Double][IGNORE_ORDER({'local': True}), INCOMPAT]
[2021-07-30T04:53:04.421Z] FAILED ../../src/main/python/join_test.py::test_sortmerge_join_with_conditionals[Decimal(18,0)][IGNORE_ORDER({'local': True})]
[2021-07-30T04:53:04.421Z] FAILED ../../src/main/python/join_test.py::test_sortmerge_join_with_conditionals[Decimal(7,3)][IGNORE_ORDER({'local': True})]
[2021-07-30T04:53:04.422Z] FAILED ../../src/main/python/join_test.py::test_sortmerge_join_with_conditionals[Decimal(7,7)][IGNORE_ORDER({'local': True})]
[2021-07-30T04:53:04.422Z] FAILED ../../src/main/python/join_test.py::test_sortmerge_join_with_conditionals[Decimal(7,-3)][IGNORE_ORDER({'local': True})]
[2021-07-30T04:53:04.422Z] FAILED ../../src/main/python/join_test.py::test_sortmerge_join_with_conditionals[Decimal(12,2)][IGNORE_ORDER({'local': True})]
[2021-07-30T04:53:04.422Z] FAILED ../../src/main/python/join_test.py::test_sortmerge_join_with_conditionals[String][IGNORE_ORDER({'local': True})]
[2021-07-30T04:53:04.422Z] FAILED ../../src/main/python/join_test.py::test_sortmerge_join_with_conditionals[Byte][IGNORE_ORDER({'local': True})]
[2021-07-30T04:53:04.422Z] FAILED ../../src/main/python/join_test.py::test_sortmerge_join_with_conditionals[Short][IGNORE_ORDER({'local': True})]
[2021-07-30T04:53:04.422Z] FAILED ../../src/main/python/join_test.py::test_sortmerge_join_with_conditionals[Integer][IGNORE_ORDER({'local': True})]
[2021-07-30T04:53:04.422Z] FAILED ../../src/main/python/join_test.py::test_sortmerge_join_with_conditionals[Long][IGNORE_ORDER({'local': True})]
[2021-07-30T04:53:04.422Z] FAILED ../../src/main/python/join_test.py::test_sortmerge_join_with_conditionals[Boolean][IGNORE_ORDER({'local': True})]
[2021-07-30T04:53:04.422Z] FAILED ../../src/main/python/join_test.py::test_sortmerge_join_with_conditionals[Date][IGNORE_ORDER({'local': True})]
```

detailed log,
```bash
[2021-07-30T04:53:04.418Z] ----------------------------- Captured stdout call -----------------------------
[2021-07-30T04:53:04.418Z] ### CPU RUN ###
[2021-07-30T04:53:04.418Z] ### GPU RUN ###
[2021-07-30T04:53:04.418Z] [31m[1m_________________ test_sortmerge_join_with_conditionals[Date] __________________[0m
[2021-07-30T04:53:04.418Z] [gw1] linux -- Python 3.8.8 /databricks/conda/envs/databricks-ml-gpu/bin/python
[2021-07-30T04:53:04.418Z] 
[2021-07-30T04:53:04.418Z] data_gen = Date
[2021-07-30T04:53:04.418Z] 
[2021-07-30T04:53:04.418Z]     [37m@ignore_order[39;49;00m(local=[94mTrue[39;49;00m)
[2021-07-30T04:53:04.418Z]     [37m@pytest[39;49;00m.mark.parametrize([33m'[39;49;00m[33mdata_gen[39;49;00m[33m'[39;49;00m, all_gen, ids=idfn)
[2021-07-30T04:53:04.418Z]     [94mdef[39;49;00m [92mtest_sortmerge_join_with_conditionals[39;49;00m(data_gen):
[2021-07-30T04:53:04.418Z]         [94mdef[39;49;00m [92mdo_join[39;49;00m(spark):
[2021-07-30T04:53:04.418Z]             left, right = create_df(spark, data_gen, [94m500[39;49;00m, [94m250[39;49;00m)
[2021-07-30T04:53:04.418Z]             [94mreturn[39;49;00m left.join(right, (left.a == right.r_a) & (left.b >= right.r_b), [33m'[39;49;00m[33mInner[39;49;00m[33m'[39;49;00m)
[2021-07-30T04:53:04.418Z] >       assert_gpu_and_cpu_are_equal_collect(do_join, conf=_sortmerge_join_conf)
[2021-07-30T04:53:04.418Z] 
[2021-07-30T04:53:04.418Z] [1m[31m../../src/main/python/join_test.py[0m:375: 
[2021-07-30T04:53:04.418Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2021-07-30T04:53:04.418Z] [1m[31m../../src/main/python/asserts.py[0m:387: in assert_gpu_and_cpu_are_equal_collect
[2021-07-30T04:53:04.418Z]     _assert_gpu_and_cpu_are_equal(func, [33m'[39;49;00m[33mCOLLECT[39;49;00m[33m'[39;49;00m, conf=conf, is_cpu_first=is_cpu_first)
[2021-07-30T04:53:04.419Z] [1m[31m../../src/main/python/asserts.py[0m:368: in _assert_gpu_and_cpu_are_equal
[2021-07-30T04:53:04.419Z]     run_on_gpu()
[2021-07-30T04:53:04.419Z] [1m[31m../../src/main/python/asserts.py[0m:362: in run_on_gpu
[2021-07-30T04:53:04.419Z]     from_gpu = with_gpu_session(bring_back, conf=conf)
[2021-07-30T04:53:04.419Z] [1m[31m../../src/main/python/spark_session.py[0m:105: in with_gpu_session
[2021-07-30T04:53:04.419Z]     [94mreturn[39;49;00m with_spark_session(func, conf=copy)
[2021-07-30T04:53:04.419Z] [1m[31m../../src/main/python/spark_session.py[0m:70: in with_spark_session
[2021-07-30T04:53:04.419Z]     ret = func(_spark)
[2021-07-30T04:53:04.419Z] [1m[31m../../src/main/python/asserts.py[0m:190: in <lambda>
[2021-07-30T04:53:04.419Z]     bring_back = [94mlambda[39;49;00m spark: limit_func(spark).collect()
[2021-07-30T04:53:04.419Z] [1m[31m/databricks/spark/python/pyspark/sql/dataframe.py[0m:697: in collect
[2021-07-30T04:53:04.419Z]     sock_info = [96mself[39;49;00m._jdf.collectToPython()
[2021-07-30T04:53:04.419Z] [1m[31m/databricks/spark/python/lib/py4j-0.10.9-src.zip/py4j/java_gateway.py[0m:1304: in __call__
[2021-07-30T04:53:04.419Z]     return_value = get_return_value(
[2021-07-30T04:53:04.419Z] [1m[31m/databricks/spark/python/pyspark/sql/utils.py[0m:110: in deco
[2021-07-30T04:53:04.419Z]     [94mreturn[39;49;00m f(*a, **kw)
[2021-07-30T04:53:04.419Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2021-07-30T04:53:04.419Z] 
[2021-07-30T04:53:04.419Z] answer = 'xro242194'
[2021-07-30T04:53:04.419Z] gateway_client = <py4j.java_gateway.GatewayClient object at 0x7f36c029b340>
[2021-07-30T04:53:04.419Z] target_id = 'o242191', name = 'collectToPython'
[2021-07-30T04:53:04.419Z] 
[2021-07-30T04:53:04.419Z]     [94mdef[39;49;00m [92mget_return_value[39;49;00m(answer, gateway_client, target_id=[94mNone[39;49;00m, name=[94mNone[39;49;00m):
[2021-07-30T04:53:04.419Z]         [33m"""Converts an answer received from the Java gateway into a Python object.[39;49;00m
[2021-07-30T04:53:04.419Z]     [33m[39;49;00m
[2021-07-30T04:53:04.419Z]     [33m    For example, string representation of integers are converted to Python[39;49;00m
[2021-07-30T04:53:04.419Z]     [33m    integer, string representation of objects are converted to JavaObject[39;49;00m
[2021-07-30T04:53:04.419Z]     [33m    instances, etc.[39;49;00m
[2021-07-30T04:53:04.419Z]     [33m[39;49;00m
[2021-07-30T04:53:04.419Z]     [33m    :param answer: the string returned by the Java gateway[39;49;00m
[2021-07-30T04:53:04.419Z]     [33m    :param gateway_client: the gateway client used to communicate with the Java[39;49;00m
[2021-07-30T04:53:04.419Z]     [33m        Gateway. Only necessary if the answer is a reference (e.g., object,[39;49;00m
[2021-07-30T04:53:04.419Z]     [33m        list, map)[39;49;00m
[2021-07-30T04:53:04.419Z]     [33m    :param target_id: the name of the object from which the answer comes from[39;49;00m
[2021-07-30T04:53:04.419Z]     [33m        (e.g., *object1* in `object1.hello()`). Optional.[39;49;00m
[2021-07-30T04:53:04.419Z]     [33m    :param name: the name of the member from which the answer comes from[39;49;00m
[2021-07-30T04:53:04.419Z]     [33m        (e.g., *hello* in `object1.hello()`). Optional.[39;49;00m
[2021-07-30T04:53:04.419Z]     [33m    """[39;49;00m
[2021-07-30T04:53:04.419Z]         [94mif[39;49;00m is_error(answer)[[94m0[39;49;00m]:
[2021-07-30T04:53:04.419Z]             [94mif[39;49;00m [96mlen[39;49;00m(answer) > [94m1[39;49;00m:
[2021-07-30T04:53:04.419Z]                 [96mtype[39;49;00m = answer[[94m1[39;49;00m]
[2021-07-30T04:53:04.419Z]                 value = OUTPUT_CONVERTER[[96mtype[39;49;00m](answer[[94m2[39;49;00m:], gateway_client)
[2021-07-30T04:53:04.419Z]                 [94mif[39;49;00m answer[[94m1[39;49;00m] == REFERENCE_TYPE:
[2021-07-30T04:53:04.419Z] >                   [94mraise[39;49;00m Py4JJavaError(
[2021-07-30T04:53:04.419Z]                         [33m"[39;49;00m[33mAn error occurred while calling [39;49;00m[33m{0}[39;49;00m[33m{1}[39;49;00m[33m{2}[39;49;00m[33m.[39;49;00m[33m\n[39;49;00m[33m"[39;49;00m.
[2021-07-30T04:53:04.419Z]                         [96mformat[39;49;00m(target_id, [33m"[39;49;00m[33m.[39;49;00m[33m"[39;49;00m, name), value)
[2021-07-30T04:53:04.419Z] [1m[31mE                   py4j.protocol.Py4JJavaError: An error occurred while calling o242191.collectToPython.[0m
[2021-07-30T04:53:04.419Z] [1m[31mE                   : org.apache.spark.SparkException: Job aborted due to stage failure: Task 1 in stage 9032.0 failed 1 times, most recent failure: Lost task 1.0 in stage 9032.0 (TID 48918) (10.2.128.6 executor driver): java.lang.IllegalStateException: Conditional joins are not supported on the GPU[0m
[2021-07-30T04:53:04.419Z] [1m[31mE                   	at org.apache.spark.sql.rapids.execution.GpuHashJoin.doJoin(GpuHashJoin.scala:642)[0m
[2021-07-30T04:53:04.419Z] [1m[31mE                   	at org.apache.spark.sql.rapids.execution.GpuHashJoin.doJoin$(GpuHashJoin.scala:611)[0m
[2021-07-30T04:53:04.419Z] [1m[31mE                   	at com.nvidia.spark.rapids.GpuShuffledHashJoinBase.doJoin(GpuShuffledHashJoinBase.scala:28)[0m
[2021-07-30T04:53:04.419Z] [1m[31mE                   	at com.nvidia.spark.rapids.GpuShuffledHashJoinBase.$anonfun$doExecuteColumnar$2(GpuShuffledHashJoinBase.scala:87)[0m
[2021-07-30T04:53:04.419Z] [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)[0m
[2021-07-30T04:53:04.419Z] [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)[0m
[2021-07-30T04:53:04.419Z] [1m[31mE                   	at com.nvidia.spark.rapids.GpuShuffledHashJoinBase.withResource(GpuShuffledHashJoinBase.scala:28)[0m
[2021-07-30T04:53:04.419Z] [1m[31mE                   	at com.nvidia.spark.rapids.GpuShuffledHashJoinBase.$anonfun$doExecuteColumnar$1(GpuShuffledHashJoinBase.scala:78)[0m
[2021-07-30T04:53:04.419Z] [1m[31mE                   	at org.apache.spark.rdd.ZippedPartitionsRDD2.compute(ZippedPartitionsRDD.scala:101)[0m
[2021-07-30T04:53:04.419Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:380)[0m
[2021-07-30T04:53:04.419Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.iterator(RDD.scala:344)[0m
[2021-07-30T04:53:04.419Z] [1m[31mE                   	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:60)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:380)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.iterator(RDD.scala:344)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:68)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:148)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.run(Task.scala:117)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$10(Executor.scala:732)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1643)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:735)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at java.lang.Thread.run(Thread.java:748)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   [0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   Driver stacktrace:[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2766)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2713)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2707)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2707)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1256)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1256)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at scala.Option.foreach(Option.scala:407)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1256)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2974)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2915)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2903)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:1029)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.SparkContext.runJobInternal(SparkContext.scala:2458)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.Collector.runSparkJobs(Collector.scala:264)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.Collector.collect(Collector.scala:299)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.Collector$.collect(Collector.scala:82)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.Collector$.collect(Collector.scala:88)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.InternalRowFormat$.collect(cachedSparkResults.scala:75)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.execution.collect.InternalRowFormat$.collect(cachedSparkResults.scala:62)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.execution.ResultCacheManager.$anonfun$getOrComputeResultInternal$1(ResultCacheManager.scala:496)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at scala.Option.getOrElse(Option.scala:189)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.execution.ResultCacheManager.getOrComputeResultInternal(ResultCacheManager.scala:495)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.execution.ResultCacheManager.getOrComputeResult(ResultCacheManager.scala:399)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.execution.ResultCacheManager.getOrComputeResult(ResultCacheManager.scala:374)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.execution.SparkPlan.executeCollectResult(SparkPlan.scala:389)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.$anonfun$collectToPython$1(Dataset.scala:3577)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3789)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$5(SQLExecution.scala:126)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:267)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withCustomExecutionEnv$1(SQLExecution.scala:104)[0m
[2021-07-30T04:53:04.420Z] [1m[31mE                   	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:852)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.withCustomExecutionEnv(SQLExecution.scala:77)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:217)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3787)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.collectToPython(Dataset.scala:3575)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at sun.reflect.GeneratedMethodAccessor138.invoke(Unknown Source)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at java.lang.reflect.Method.invoke(Method.java:498)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:380)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at py4j.Gateway.invoke(Gateway.java:295)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at py4j.commands.CallCommand.execute(CallCommand.java:79)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at py4j.GatewayConnection.run(GatewayConnection.java:251)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at java.lang.Thread.run(Thread.java:748)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   Caused by: java.lang.IllegalStateException: Conditional joins are not supported on the GPU[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.sql.rapids.execution.GpuHashJoin.doJoin(GpuHashJoin.scala:642)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.sql.rapids.execution.GpuHashJoin.doJoin$(GpuHashJoin.scala:611)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at com.nvidia.spark.rapids.GpuShuffledHashJoinBase.doJoin(GpuShuffledHashJoinBase.scala:28)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at com.nvidia.spark.rapids.GpuShuffledHashJoinBase.$anonfun$doExecuteColumnar$2(GpuShuffledHashJoinBase.scala:87)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at com.nvidia.spark.rapids.GpuShuffledHashJoinBase.withResource(GpuShuffledHashJoinBase.scala:28)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at com.nvidia.spark.rapids.GpuShuffledHashJoinBase.$anonfun$doExecuteColumnar$1(GpuShuffledHashJoinBase.scala:78)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.rdd.ZippedPartitionsRDD2.compute(ZippedPartitionsRDD.scala:101)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:380)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.iterator(RDD.scala:344)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:60)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:380)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.iterator(RDD.scala:344)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:68)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:148)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.run(Task.scala:117)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$10(Executor.scala:732)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1643)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:735)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)[0m
[2021-07-30T04:53:04.421Z] [1m[31mE                   	... 1 more[0m
```

--- a/shims/spark311db/src/main/scala/com/nvidia/spark/rapids/shims/spark311db/GpuSortMergeJoinExec.scala
+++ b/shims/spark311db/src/main/scala/com/nvidia/spark/rapids/shims/spark311db/GpuSortMergeJoinExec.scala
@@ -78,14 +78,17 @@ class GpuSortMergeJoinMeta(
       throw new IllegalStateException(s"Cannot build either side for ${join.joinType} join")
     }
     val Seq(leftChild, rightChild) = childPlans.map(_.convertIfNeeded())
-    GpuShuffledHashJoinExec(
+    val joinExec = GpuShuffledHashJoinExec(
       leftKeys.map(_.convertToGpu()),
       rightKeys.map(_.convertToGpu()),
       join.joinType,
       GpuJoinUtils.getGpuBuildSide(buildSide),
-      condition.map(_.convertToGpu()),
+      None,
       leftChild,
       rightChild)
+    // The GPU does not yet support conditional joins, so conditions are implemented
+    // as a filter after the join when possible.
+    condition.map(c => GpuFilterExec(c.convertToGpu(), joinExec)).getOrElse(joinExec)
   }
 
   /**

